### DISCUSSION TITLE
once again just got to say this is the best software ever

### DISCUSSION THREAD

**Discussion by imgonnablowthisplaceup** [ReagentX/imessage-exporter#103] [General]
all I gots to say 🙏

**Comment by warptrail** [comment]
I must second this notion, I have been searching for a way to search through my mountains of text data. Apple engineers would be wise to study this repo.

**Comment by craigjperry2** [comment]
Just wanted to add my appreciation. I was exporting a 7 year thread, it's just under 2Gb (photos etc) and it exported in UNDER 1 SECOND!

```
20:17:23 ~ ❯ imessage-exporter -f html -c clone -t 'REDACTED,REDACTED,REDACTED'
Building cache...
  [1/4] Caching chats...
  [2/4] Caching chatrooms...
  [3/4] Caching participants...
  [4/4] Caching tapbacks...
Cache built!
Filtering for 2 handles across 7 chatrooms...
Estimated export size: 1.72 GB
Exporting to /Users/craig/imessage_export as html...
  [0s] [####################] 12,375/12,375 (27,183.4512/s, ETA: 0s)
Writing HTML footers...
Done!
```

**Comment by tylerseymour** [comment]
Yes, THANK YOU @ReagentX !!  

My phone had inched up to 110+ GB of conversation history, but I couldn't bare the thought of deleting 17 years worth of memories.  

I can't believe how quickly it works to boot!  110GB extracted and structured in about 20 seconds.

**Comment by memphis-10** [comment]
I'm here to add the same: THANK YOU! This is legendary work. I know it wasn't easy, so very much appreciated!!
